### ISSUE TITLE
Different random seeds can cause differences in arithmetic precision

### ISSUE THREAD

**Issue by chadwick-yu** [pytorch/pytorch#132989]
### 🐛 Describe the bug

Code Block1
```python
torch.manual_seed(1337)  #torch.manual_seed(42)
B,T,C = 4,8,2 # batch, time, channels
x = torch.randn(B,T,C)
#version1
xbow = torch.zeros((B,T,C))
for b in range(B):
    for t in range(T):
        xprev = x[b,:t+1] # (t,C)
        xbow[b,t] = torch.mean(xprev, 0)
```
Code Block 2
``` python
# version 2
wei = torch.tril(torch.ones(T, T)) 
wei = wei / wei.sum(1, keepdim=True)        
xbow2 = wei @ x # (B, T, T) @ (B, T, C) ----> (B,T,C)  
torch.allclose(xbow, xbow2)
``` 
The random seed 1337 and the random seed 42 cause a difference in the results of the above operations, and ultimately the output of the allclose function is completely different
![image](https://github.com/user-attachments/assets/5bbbe8de-a158-4df2-8a38-a43b6950e4b3)
![image](https://github.com/user-attachments/assets/d23f9ca5-3a42-49e5-b4d8-384b5bab4a97)
As shown in the figure

### Versions

the two code blocks run in the colab not in the colab pro

**Comment by albanD** [comment]
I'm curious if this could be due to https://pytorch.org/docs/stable/notes/numerical_accuracy.html#batched-computations-or-slice-computations ?

**Comment by nobody81127** [comment]
I applied a floating-point error analysis algorithm to calculate the error bounds for the result of xbow, and I found that the result of xbow2 falls within this range. Therefore, I believe this is indeed a floating-point error, likely caused by differences in the order of operations. Therefore, I suggest closing this issue. @chadwick-yu @albanD
